### PR TITLE
Chore/use core lib

### PR DESCRIPTION
**Pull request by jasoniangreen**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -12,6 +12,13 @@ var preprocess = require('gulp-preprocess');
 var preprocessify = require('preprocessify');
 var merge = require('merge-stream');
 
+gulp.task('newcore', function() {
```
Comment (jasoniangreen) [inline gulpfile.js]: Basically to use the new core while changing as little as possible, I decided to generate the core folder from the new core lib and then let everything run as it has been doing so before.

File: .gitignore [hunk 2]
```diff
@@ -2,5 +2,7 @@
 literallycanvas/
 *.sublime-workspace
 node_modules/
+core/
```
Comment (jasoniangreen) [inline .gitignore]: Core is now generated in `prepublish`.

File: .gitignore [hunk 3]
```diff
@@ -2,5 +2,7 @@
...
+core/fontmetrics.js
```
Comment (irskep) [inline .gitignore]: Unfortunately the main `fontmetrics` package has an XSS bug, so this should be reverted. I don't trust it to continue to be correct.

File: README.md [hunk 4]
```diff
@@ -1,8 +1,8 @@
-Literally Canvas v0.5.2
+Literally Canvas v1.0.0
```
[the remark this answers is not in the document]
Comment (irskep) [reply README.md]: Sounds good

File: README.md [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
 =======================
 
 Literally Canvas is an extensible, open source (BSD-licensed), HTML5 drawing
-widget. Its only dependency is [React.js](http://facebook.github.io/react/).
+widget. Its only dependencies are [React.js](http://facebook.github.io/react/) and it's core lib which has been split up into a separate module [literallycanvas-core](https://github.com/literallycanvas/literallycanvas-core/).
```
Comment (irskep) [inline README.md]: *its

File: gulpfile.js (hunk 1 above)
Comment (irskep) [reply gulpfile.js]: This seems like a good compromise.

**Review by irskep — CHANGES_REQUESTED** [review]
Needs the non-XSS-vulnerable version of `fontmetrics`

**Comment by jasoniangreen** [comment]
@irskep I really want to get the new core being used with the old codebase first. To do so with as little change as possible, I am generating the core folder using the code in the new core lib. I like incremental change :) It all seems to be working for me, the demos work, even if they are a bit glitchy in terms of layout (they were before I did anything, so I can't count that as a fail :) )

File: .gitignore (hunk 3 above)
Comment (jasoniangreen) [reply .gitignore]: Oh I see.... but I have the same version you had in the core, so it should be pulling in [this one](https://github.com/literallycanvas/literallycanvas-core/blob/master/src/fontmetrics.js) when you run buildcore.

File: README.md (hunk 5 above)
Comment (jasoniangreen) [reply README.md]: Doh!

File: .gitignore (hunk 3 above)
Comment (irskep) [reply .gitignore]: Oh, ok! Then can't we just remove it from the repo?

Comment (irskep) [reply .gitignore]: I mean rather than adding it to gitignore, since it's gone anyway.

**Review by irskep — APPROVED** [review]

File: .gitignore (hunk 3 above)
Comment (jasoniangreen) [reply .gitignore]: I'll try again, for some reason something kept tracking it's changes, despite the fact that the whole core folder is supposed to be ignored. I'll check again.

Comment (jasoniangreen) [reply .gitignore]: Ok, seems fine now.... just some git strangeness.

**Comment by irskep** [comment]
Added you as a collaborator so you can merge on your own!

**Comment by jasoniangreen** [comment]
Thanks, but who does the publish? I setup circle CI with semantic-release in the core project, I'm happy to have a crack at it on this one if you like.

**Pull request merged**